server/webapp/WEB-INF/rails.new/app/assets/javascripts/pipeline_configs/models/tasks.js

```diff
@@ -47,6 +59,14 @@ define(['mithril', 'lodash', 'string-plus', './model_mixins', './argument', './r
     this._attributesToJSON = function () {
       throw new Error("Subclass responsibility!");
     };
+
+    this.hasOnCancelTask = function () {
+      return ! _.isNil(this.onCancelTask);
```

Comment:
Did not know about `isNil`, can you get rid of `isUndefinedOrNull` in `string-plus.js` (or change impl to `isUndefinedOrNull: _.isNil`)